bindings/rust/s2n-tls/src/config.rs

```diff
@@ -36,6 +36,14 @@ impl Config {
     /// Returns a Config object with pre-defined defaults.
     ///
     /// Use the [`Builder`] if custom configuration is desired.
+    ///
+    /// # Warning
+    ///
+    /// By default, the newly created Config uses the default security policy.
```

Comment:
```suggestion
    /// By default, the newly created Config uses the default security policy. This policy changes across library versions and could break connections.
```
Should this short version of the warning mention that the policy changes? Unless I already knew about the default policy/had read the other documentation, I probably wouldn't understand why this was a warning.

Reply:
Idk, the changing isn't the only issue. The default policy might also just not have the options they're expecting / wanting. Like, at the moment it still has 1.0 and most customers probably don't want 1.0.

I was hoping I just needed to point out policies even exist so customers might read the other documentation :/

Reply:
Ya, if we think the default is going to remain stuck at 1.0, then I wonder if it would be worth exploring the option of totally removing the default from the bindings?

But still think we should merge this PR, since the new warning is definitely better than what we have now 😄